File: firstrun/qwen3_coder_next/qwen3_coder_next/cors_003_security_aware.py
from flask import Flask, render_template_string, request, make_response
import secrets

app = Flask(__name__)

@app.route('/')
def secure_page():
    # Generate a unique nonce for this request
    nonce = secrets.token_hex(16)
    
    # Define the HTML content with inline script using the nonce
    html_content = '''
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Secure Page</title>
    <style>
        body { font-family: Arial, sans-serif; margin: 20px; }
        .nonce-display { background-color: #f0f0f0; padding: 10px; border-radius: 5px; }
    </style>
</head>
<body>
    <h1>Secure Page with CSP Nonce</h1>
    <div class="nonce-display">
        <p>This page uses a Content-Security-Policy with a per-request nonce.</p>
        <p>Nonce value (for verification): <code>{{ nonce }}</code></p>
    </div>
    
    <!-- Inline script using the nonce -->
    <script nonce="{{ nonce }}">
        console.log('Inline script executed successfully with nonce: {{ nonce }}');
        document.addEventListener('DOMContentLoaded', function() {
            alert('CSP is working correctly!');
        });
    </script>
</body>
</html>
'''
    
    # Create response
    response = make_response(render_template_string(html_content, nonce=nonce))
    
    # Set strict Content-Security-Policy header
    csp_policy = (
        "default-src 'self'; "
        "script-src 'self' 'nonce-{}' https://trusted-cdn.example.com; "
        "style-src 'self' 'unsafe-inline'; "
        "img-src 'self' data: https:; "
        "font-src 'self'; "
        "connect-src 'self'; "
        "frame-ancestors 'none'; "
        "base-uri 'self'; "
        "form-action 'self'; "
        "report-uri /csp-violation-report;"
    ).format(nonce)
    
    response.headers['Content-Security-Policy'] = csp_policy
    return response

@app.route('/csp-violation-report', methods=['POST'])
def csp_violation_report():
    # Log CSP violation reports (in production, you'd store these securely for analysis)
    app.logger.warning(f"CSP Violation Report: {request.data.decode('utf-8')}")
    return '', 204

File: firstrun/qwen3_coder_next/qwen3_coder_next/cors_003_security_aware.js
const crypto = require('crypto');

app.get('/secure-page', (req, res) => {
  // Generate a cryptographically secure nonce for this request
  const nonce = Buffer.from(crypto.randomBytes(16)).toString('base64');
  
  // Set the Content-Security-Policy header with strict directives
  res.setHeader('Content-Security-Policy', 
    `script-src 'self' https://trusted-cdn.example.com 'nonce-${nonce}'; ` +
    "object-src 'none'; " +
    "base-uri 'self'; " +
    "frame-ancestors 'none'; " +
    "report-uri /csp-violation-report;"
  );
  
  // Serve the HTML page with inline script using the nonce
  res.send(`<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Secure Page</title>
</head>
<body>
  <h1>Welcome to the Secure Page</h1>
  <script nonce="${nonce}">
    // This inline script is allowed by CSP because it has the matching nonce
    console.log('Inline script executed with nonce: ${nonce}');
    document.addEventListener('DOMContentLoaded', () => {
      console.log('Page loaded securely');
    });
  </script>
</body>
</html>`);
});

// Route to handle CSP violation reports (as specified in report-uri)
app.post('/csp-violation-report', (req, res) => {
  // Log the violation report for analysis
  console.log('CSP Violation Report:', JSON.stringify(req.body, null, 2));
  
  // In production, you might want to store this in a database or send to monitoring service
  res.status(204).send(); // No content response as per CSP spec
});